src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/ElasticSearchAnalysisGraph.java

```diff
@@ -150,8 +157,40 @@ public void construct() {
     hotNodeClusterRca.addTag(TAG_LOCUS, LOCUS_MASTER_NODE);
     hotNodeClusterRca.addAllUpstreams(Collections.singletonList(hotJVMNodeRca));
 
+    /* Queue Rejection RCAs
+     */
+    // TODO: Refactor this monolithic function
+    Metric threadpool_RejectedReqs = new ThreadPool_RejectedReqs(EVALUATION_INTERVAL_SECONDS);
+    threadpool_RejectedReqs.addTag(TAG_LOCUS, LOCUS_DATA_MASTER_NODE);
+    addLeaf(threadpool_RejectedReqs);
+
+    // Node level queue rejection RCA
+    QueueRejectionRca queueRejectionNodeRca = new QueueRejectionRca(RCA_PERIOD, threadpool_RejectedReqs);
+    queueRejectionNodeRca.addTag(TAG_LOCUS, LOCUS_DATA_MASTER_NODE);
+    queueRejectionNodeRca.addAllUpstreams(Collections.singletonList(threadpool_RejectedReqs));
+
+    // Cluster level queue rejection RCA
+    QueueRejectionClusterRca queueRejectionClusterRca = new QueueRejectionClusterRca(RCA_PERIOD, queueRejectionNodeRca);
+    queueRejectionClusterRca.addTag(TAG_LOCUS, LOCUS_MASTER_NODE);
+    queueRejectionClusterRca.addAllUpstreams(Collections.singletonList(queueRejectionNodeRca));
+
+    // Queue Health Decider
+    QueueHealthDecider queueHealthDecider = new QueueHealthDecider(EVALUATION_INTERVAL_SECONDS, 12, queueRejectionClusterRca);
+    queueHealthDecider.addTag(TAG_LOCUS, LOCUS_MASTER_NODE);
+    queueHealthDecider.addAllUpstreams(Collections.singletonList(queueRejectionClusterRca));
+
```

Comment:
can we extract out the code here in a method of its own as in `constructShardResourceUsageGraph` ?